usr.bin/look/look.c

```diff
@@ -132,22 +137,38 @@ main(int argc, char *argv[])
 
 	match = 1;
 
-	do {
-		if ((fd = open(file, O_RDONLY, 0)) < 0 || fstat(fd, &sb))
+	cap_rights_init(&rights, CAP_MMAP_R, CAP_READ, CAP_FSTAT);
+	nfiles = argc > 1 ? argc - 1 : argc;
+	if ((fds = malloc(nfiles * sizeof(int))) == NULL)
+		err(2, NULL);
+	for (size_t idx = 0; idx < nfiles; file = argv[idx++]) {
+		if ((fds[idx] = open(file, O_RDONLY, 0)) < 0)
+			continue;
+		if (caph_rights_limit(fds[idx], &rights) != 0)
+			err(2, "unable to limit rights for %s", file);
+	}
+
+	caph_cache_catpages();
+	if (caph_enter() != 0)
+		err(EXIT_FAILURE, "failed to enter capability mode");
+
+	for (size_t idx = 0; idx < nfiles; file = argv[idx++]) {
+		if (fstat(fds[idx], &sb))
```

Comment:
I think this is still wrong. I have not run this code, so I might be wrong here.

The open(2) returns `-1` for failed open. We ignore actual error codem which is in `errno`. Then we pass -1 to `fstat(2)`. In result instead of meaningful error like "File doesn't exists", "No access" ect. we provide user with a same error "Invalid file descriptor" when the `fstat(2)` fails.